From 194c2a1538b135b6e20616f619c0c751e7702fbf Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Wed, 21 Nov 2012 10:49:49 +0800
Subject: [PATCH 096/162] net/gianfar: Revert "gianfar: Add HW VLAN accel
 disable driver param"

The commit 87c288c6 (gianfar: do vlan cleanup) adds the support
to turn on/off rx/tx vlan accel via ethtool. So this function seems
redundant, and just drop it.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/freescale/gianfar.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/freescale/gianfar.c b/drivers/net/ethernet/freescale/gianfar.c
index 487848d..1cdcb2f 100644
--- a/drivers/net/ethernet/freescale/gianfar.c
+++ b/drivers/net/ethernet/freescale/gianfar.c
@@ -121,11 +121,9 @@
 
 const char gfar_driver_version[] = "1.3";
 static struct gfar_recycle_cntxt *gfar_global_recycle_cntxt;
-static bool hw_vlan_enabled = 1;
 static int tx_napi_enabled = 1;
 static int tx_napi_weight = GFAR_DEV_TX_WEIGHT;
 static int rx_napi_weight = GFAR_DEV_RX_WEIGHT;
-module_param(hw_vlan_enabled, bool, S_IRUGO);
 module_param(tx_napi_enabled, bool, S_IRUGO);
 module_param(tx_napi_weight, int, S_IRUGO);
 module_param(rx_napi_weight, int, S_IRUGO);
@@ -1339,7 +1337,7 @@ static int gfar_probe(struct platform_device *ofdev)
 			NETIF_F_RXCSUM | NETIF_F_HIGHDMA;
 	}
 
-	if ((priv->device_flags & FSL_GIANFAR_DEV_HAS_VLAN) && hw_vlan_enabled) {
+	if (priv->device_flags & FSL_GIANFAR_DEV_HAS_VLAN) {
 		dev->hw_features |= NETIF_F_HW_VLAN_TX | NETIF_F_HW_VLAN_RX;
 		dev->features |= NETIF_F_HW_VLAN_RX;
 	}
-- 
1.7.9.7

